test/e2e/performance_test.go

```diff
@@ -0,0 +1,371 @@
+// Copyright 2019 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"flag"
+	"fmt"
+	"net/url"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+	"unsafe"
+
+	"golang.org/x/exp/rand"
+
+	v1 "k8s.io/api/core/v1"
+	networkv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	seed                            uint64 = 0xA1E47 // Use a specific rand seed to make the generated workloads always same
+	performanceAppLabel                    = "antrea-performance-test"
+	performanceTestPodPrefix               = performanceAppLabel + "-pod"
+	podsConnectionNetworkPolicyName        = performanceTestPodPrefix + "pods.ingress"
+	workloadNetworkPolicyName              = performanceTestPodPrefix + "workloads.ingress"
+	abImage                                = "tangweiqiang/ab"
+	abName                                 = "ab"
+)
+
+var (
+	performanceNginxPodName = randPodName(performanceTestPodPrefix + "-nginx-")
+	performanceABPodName    = randPodName(performanceTestPodPrefix + "-ab-")
+
+	httpConcurrency = flag.Uint("performance.http.concurrency", 16, "Number of multiple requests to make at a time")
+	realizeTimeout  = flag.Duration("performance.realizeTimeout", 5*time.Minute, "Timeout of the realization of network policies")
+)
+
+func BenchmarkBareHTTPRequest100000(b *testing.B) {
+	withPerformanceTestSetup(func(data *TestData) { httpRequest(100000, 0, data, b) }, b)
```

Comment:
rather than multiplying the number of benchmark functions, I think we should make the parameters ("times", "amount") flags for the test, and have reasonable defaults. What do you think?

Reply:
I grouped benchmark test of same category and addded two customized tests, how do you feel about this?

Reply:
I think it's not bad as it is, being able to run customized test is good. We can always adjust in the future.